### PR TITLE
nh: init at 3.4.12

### PR DESCRIPTION
**Pull request by drupol**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by viperML** [comment]
> @viperML Are you willing to be added in the maintainers for it ?

Ok, feel free to add me on this PR or I will do a follow-up PR.

---

The current nh version is `3.4.12`, but I didn't use git tags (just in the Cargo.toml). I just pushed a tag https://github.com/viperML/nh/tree/v3.4.12 so it can be consumed from fetchFromGitHub easily.

**Review by NobbZ — APPROVED** [review]
Also, please use the tag as created by Viper recently

**Review by NobbZ — CHANGES_REQUESTED** [review]
Also, please use the tag as created by Viper recently

**Comment by viperML** [comment]
lgtm, thanks

File: pkgs/by-name/nh/nh/package.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+rustPlatform.buildRustPackage {
+  inherit version;
+  pname = "nh";
+
+  src = fetchFromGitHub {
+    owner = "ViperML";
+    repo = "nh";
+    rev = version;
+    hash = "sha256-V5TQ/1loQnegDjfLh61DxBWEQZivYEBq2kQpT0fn2cQ=";
+  };
```
Comment (NobbZ) [inline pkgs/by-name/nh/nh/package.nix:26]: Seems as if you messed up the `rev`.

I think best would be `refs/tags/v${version}`.

```text
source> building '/nix/store/jyqhkp4fdv8qa43ghkb3059c6v8cswb8-source.drv'
source>
source> trying https://github.com/ViperML/nh/archive/3.4.12.tar.gz
source>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
source>                                  Dload  Upload   Total   Spent    Left  Speed
source>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
source>   0    14    0     0    0     0      0      0 --:--:--  0:00:03 --:--:--     0
source> curl: (22) The requested URL returned error: 404
source> error: cannot download source from any mirror
```

**Review by NobbZ — CHANGES_REQUESTED** [review]
Please fix the sources

File: pkgs/by-name/nh/nh/package.nix (hunk 1 above)
Comment (drupol) [reply pkgs/by-name/nh/nh/package.nix:26]: Oops. Going to fix that. Thx !

File: pkgs/by-name/nh/nh/package.nix [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    installShellCompletion completions/*
+  '';
+
+  postFixup = ''
+    wrapProgram $out/bin/nh \
+      --prefix PATH : ${lib.makeBinPath runtimeDeps} \
+      ${lib.optionalString use-nom "--set-default NH_NOM 1"}
+  '';
+
+  cargoHash = "sha256-1JcHgfwM3Gv7amyxGkwBI4sgWVlq4z02Pv6EkX/I4W0=";
```
Comment (NobbZ) [inline pkgs/by-name/nh/nh/package.nix:50]:
```suggestion
  cargoHash = "sha256-Ul4DM8WmKvKG32zBXzpdzHZknpTQAVvrxFcEd/C1buA=";
```

Can you please make it a habbit to always reset all hashes when doing small changes to the `src`? Even if you *think* that they might not cause any changes, please always do!
FODs already realized can be nastly…

**Review by NobbZ — CHANGES_REQUESTED** [review]

File: pkgs/by-name/nh/nh/package.nix (hunk 2 above)
Comment (drupol) [reply pkgs/by-name/nh/nh/package.nix:50]: Oops... you're right, sorry.

Comment (drupol) [reply pkgs/by-name/nh/nh/package.nix]: Job done.

**Review by NobbZ — APPROVED** [review]

**Pull request merged**